### ISSUE TITLE
UpdateBy without entity available?

### ISSUE THREAD

**Issue by lk-code** [TanvirArjel/EFCore.GenericRepository#48]
hi,

Is there an update method without having to have an entity? Something like this:

from this C# code
```
repository.Update(x => x.Id == 4, x => x.Name = "Updated Value");
```

to this sql
```
UPDATE table SET Name = 'Updated Value' WHERE Id = 4;
```

or update multiple entities like this
```
repository.Update(x => x.OtherValue is null, x => x.Name = "Updated Value");
```
to this sql
```
UPDATE table SET Name = 'Updated Value' WHERE OtherValue = NULL;
```

**Comment by NotAsea** [comment]
How even EF can know what entity you're refering to so it can update that table. Remember this lib only wrap around EFCore and doing some chore work so we dont have whole project/folder just for repository pattern, your question should belong to efcore repo, even then i just want to ask what real world use case for your problem

**Comment by lk-code** [comment]
we can use
```
repository.Update<Entity>(x => x.Id == 4, x => x.Name = "Updated Value");
```

at the moment i have to load the whole entity every time with EF. then id need to change the values and save the entity again.

With my new way, we could save a database query and mapping into the entity-model.
There would only be one generated SQL. No model mapping, etc. I think that would be a good way to optimize performance ;)

**Comment by NotAsea** [comment]
Are you refering to batch update, if so, yes EFcore 7+ does support it, and this lib has it as well, just use  [repository.ExcuteUpdateAsync](https://github.com/TanvirArjel/EFCore.GenericRepository/blob/cf5df72e31a252dd7431593b04f505004c657d87/src/TanvirArjel.EFCore.GenericRepository/Repository.cs#L304 ).
```cs 
await repository.ExcuteUpdateAsync<Entity>(
    x => x.ThisConditionForUpdate, 
    setter => setter.SetProperty(x=> x.ThisSupposeToBeUpdateProperty, whateverValue)
       //can chain SetProperty to update multiple field
);
```
 For support EFcore 6 and below you need to use other library like Z efcore extension to do batch update as it not native support by efcore.
tl;dr: your first code example does throw me around, like i even thougt you ask for update whatever field in database

**Comment by lk-code** [comment]
aah yeah, thats exactly what im looking for :D thanks :)
